### PR TITLE
Create composer cookbook, along with chefspec testing

### PR DESCRIPTION
**Pull request by fahlmant**
Rake fails on a library file, but passes on everything else.

### CODE REVIEW

**Comment by fahlmant** [comment]
@osuosl-cookbooks/chefs

**Comment by michaelcullum** [comment]
Where does composer come from?

**Comment by fahlmant** [comment]
Hi @michaelcullum. The composer cookbook is what is installing the composer package in this recipe. We are relying on the upstream cookbook to handle installation. We will most likely lock the version so we control the update. This change will not affect phpbb at all. I am writing the recipe for another project that needs composer.

**Comment by michaelcullum** [comment]
We use composer all over the place though, we just don't have it globally accessible but instead have a composer.phar in each project.

Essentially we'd like to use this (also this change directly affects us as running `composer` would suddenly refer to this instead of the `composer.phar` file in those directories so we kind of have to use it or change our stuff). But we'd only want to use it if it was set to update regularly (you should run a composer update at least once every 30 days absolute maximum, but as good practice composer should always be up to date when used so having chef automatically running `composer self-update` regularly or however it keeps up to date would be good, normally one simply runs `composer self-update` before running composer commands but of course chef will overwrite that).

**Comment by fahlmant** [comment]
@michaelcullum This recipe isn't in use yet, so when merged, this will not affect your composer unless we explicitly make this recipe run on your nodes.

As for the update method, we could easily have it run an update on a schedule. I can look into enabling this for your nodes if you'd like.

**Comment by ramereth** [comment]
@michaelcullum looks like we're only enabling installing the composer "binary" in /usr/local/bin. There is a recipe to run self update, but we're not enabling it here. This also doesn't affect other users' composer install unless we specify it.

**Comment by michaelcullum** [comment]
@fahlmant Ah sorry, I assumed our use of osl-php would pull this in too.

@ramereth Sure, but having a file `composer` in the path (`usr/local/bin`) would mean running `composer` in the cli in various places where we do would run that composer binary instead of running `php composer.phar` on the composer.phar in the current directory unless I'm mistaken.

**Comment by ramereth** [comment]
@michaelcullum that depends if /usr/local/bin is in the ``PATH`` where the script runs. You could also ensure that's not included or use an absolute path for running composer to the binary you want to run.

**Comment by fahlmant** [comment]
@osuosl-cookbooks/chefs Ready for review. Rubocop fails on a library definition, but temporarily removing it shows no issues with the changes I've made.

File: .kitchen.yml [hunk 1]
```diff
@@ -23,6 +23,11 @@ suites:
...
       - recipe[osl-php::apc]
     attributes:
-      osl-php:
-        apc:
-          shm_size: 128M
+      php:
+        opcache:
+          opcache.interned_strings_buffer: 20M
+          opcache.max_accelerated_files: 10000
+          opcache.memory_consumption: 200M
```
Comment (ramereth) [inline .kitchen.yml]: Why the change here?

File: spec/composer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+require_relative 'spec_helper'
+
+describe 'osl-php::composer' do
+  cached(:chef_run) do
+    ChefSpec::SoloRunner.new(CENTOS_7_OPTS).converge(described_recipe)
```
Comment (ramereth) [inline spec/composer_spec.rb]: You should also test against CentOS 6 here.

File: .kitchen.yml (hunk 1 above)
Comment (fahlmant) [inline .kitchen.yml]: Really not even sure where this came from

File: attributes/composer.rb [hunk 3]
```diff
@@ -0,0 +1 @@
+default['composer']['url'] = 'https://getcomposer.org/download/1.2.1/composer.phar'
```
Comment (ramereth) [inline attributes/composer.rb]: Please move this to the composer recipe and use ``node[foo]`` instead of default.

**Comment by fahlmant** [comment]
@DesertBeagle In the composer recipe that is included

File: attributes/composer.rb (hunk 3 above)
Comment (ramereth) [inline attributes/composer.rb]: You use ``node.default['foo']`` in the recipe.

**Comment by fahlmant** [comment]
@ramereth ready for final review

**Comment by fahlmant** [comment]
!bump patch ~

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a patch-level version bump to v0.1.3.  Have a nice day!
